File: A1/A1-037/A1-037.cpp
#include<iostream>
using namespace std;

// Roman number on each digits
string THOUSANDS[4] =  {"", "M", "MM", "MMM"};
string HUNDREDS[10] =  {"", "C", "CC", "CCC", "CD",
                        "D", "DC", "DCC", "DCCC", "CM"};
string TENS[10]     =  {"", "X", "XX", "XXX", "XL",
                        "L", "LX", "LXX", "LXXX", "XC"};
string ONES[10]     =  {"", "I", "II", "III", "IV",
                        "V", "VI", "VII", "VIII", "IX"};

int main()
{
    // This makes std::cin and std::cout faster
    ios_base::sync_with_stdio(false); cin.tie(0);

    // Input
    int num;
    cin >> num;
    // Output
    cout << THOUSANDS[num / 1000];
    num %= 1000;
    cout << HUNDREDS[num / 100];
    num %= 100;
    cout << TENS[num / 10];
    num %= 10;
    cout << ONES[num];
    return 0;
}

File: A1/A1-037/A1-037.py
# --------------------------------------------------
# File Name : A1-037.py
# Problem   : Roman Number
# Author    : Worralop Srichainont
# Date      : 2026-03-13
# --------------------------------------------------

# Roman number on each digits
THOUSANDS = ["", "M", "MM", "MMM"]
HUNDREDS = ["", "C", "CC", "CCC", "CD", "D", "DC", "DCC", "DCCC", "CM"]
TENS = ["", "X", "XX", "XXX", "XL", "L", "LX", "LXX", "LXXX", "XC"]
ONES = ["", "I", "II", "III", "IV", "V", "VI", "VII", "VIII", "IX"]

# Input number and extract digits
digits = [0, 0, 0, 0]
for digit in input().strip():
    digits.append(int(digit))

# Construct roman number
roman_number = (
    THOUSANDS[digits[-4]] + HUNDREDS[digits[-3]] + TENS[digits[-2]] + ONES[digits[-1]]
)

# Output
print(roman_number)
